tests/lib/rules/validate-set-cookie-header/tests.ts

```diff
@@ -0,0 +1,133 @@
+/* eslint sort-keys: 0, no-undefined: 0 */
+
+import { RuleTest } from '../../../helpers/rule-test-type'; // eslint-disable-line no-unused-vars
+import * as ruleRunner from '../../../helpers/rule-runner';
+import { getRuleName } from '../../../../src/lib/utils/rule-helpers';
+const ruleName = getRuleName(__dirname);
+
+// Headers.
+const setCookie = (fields) => {
+    return { 'Set-Cookie': fields };
+};
+
+// Headers that will pass.
+const doubleQuotedValueHeader = setCookie(`cookieName="cookieValue"; Secure; HttpOnly`);
+const standardHeader = setCookie(`cookieName=cookieValue; Secure; HttpOnly`);
+const starnderHeaderLowerCase = setCookie(`cookieName=cookieValue; secure; httponly`);
+const standardHeaderWithSecurePrefix = setCookie(`__Secure-ID=123; Secure; Domain=example.com; HttpOnly`);
+const standardHeaderWithHostPrefix = setCookie(`__Host-ID=123; Secure; Path=/; HttpOnly`);
+
+// Headers that will fail.
+const noNameValueStringHeader = setCookie(`Max-Age=0; Secure; HttpOnly`);
+const invalidAttributeHeader = setCookie(`cookieName=cookieValue; MaxAge=0; Secure; HttpOnly`);
+const noSecureHeader = setCookie(`cookieName=cookieValue; HttpOnly`);
+const noHttpOnlyHeader = setCookie(`cookieName=cookieValue; Secure`);
+
+const invalidNameHeader = setCookie(`"cookieName"=cookieValue; Secure; HttpOnly`);
+const invalidValueHeader = setCookie(`cookieName=cookie value; Secure; HttpOnly`);
+const invalidDateFormatHeader = setCookie(`cookieName=cookieValue; expires=Wed, 31-Dec-97 23:59:59 GMT; Secure; HttpOnly`);
+const trailingSemicolonHeader = setCookie(`cookieName=cookieValue;`);
+const multipleErrorsHeader = setCookie(`"cookieName"=cookie value`);
+
+const noPathHostPrefixHeader = setCookie(`__Host-id=1; Secure; HttpOnly`);
+const hasDomainHostPrefixHeader = setCookie(`__Host-id=1; Secure; Path=/; domain=example.com; HttpOnly`);
+
+
+// Error messages.
+const messages = (cookieName: string = 'cookiename'): { [key: string]: string } => {
```

Comment:
@alrra This is because `cookieName` can be different for a certain message, and it seems redundant to repeat itself when `cookieName` is the only difference. So the function is needed to interpolate the `cookieName`.